### PR TITLE
Implement #109 adding action types file

### LINKED ISSUE #109: Add action typing
I'm looking to get your action first-class supported by https://github.com/krzema12/github-workflows-kt, a Kotlin DSL to write GitHub Action workflows.

They came up with a way to reduce operational load when keeping library's action wrappers in sync with actions' inputs and outputs. The solution includes onboarding https://github.com/krzema12/github-actions-typing. It's as easy as adding an extra YAML file to your repository root, and adding a simple GitHub workflow that validates this new file. Thanks to this, the code generator in the Kotlin DSL can fetch typing info provided by you instead of them, which has a number of benefits. It has no negative effects on current action consumers, they continue to use the action via regular GitHub API, as if the file wasn't there.

In this feature request, I would like to ask you if you're open to introducing such typings in your action. You wouldn't be first - there're already other actions using it, like e.g.: https://github.com/Vampire/setup-wsl or https://github.com/ReactiveCircus/android-emulator-runner (full list [here](https://github.com/krzema12/github-actions-typing/network/dependents)).

If your answer is "yes", feel free to either add it yourself, or let me know - me or some of the library contributors would be happy to post a PR. They're also open to any kind of questions and feedback.

### PR DESCRIPTION
**Pull request by Vampire**
Fixes #109

### CODE REVIEW

**Comment by Vampire** [comment]
@microsoft-github-policy-service agree

File: action-types.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  vs-version:
+    type: string
+  vs-prerelease:
+    type: boolean
+  msbuild-architecture:
+    type: enum
+    name: Architecture
+    allowed-values:
+    - x86
+    - x64
```
Comment (timheuer) [inline action-types.yml:14]: arm64 is also a valid option

**Review by timheuer — CHANGES_REQUESTED** [review]

File: action-types.yml (hunk 1 above)
Comment (Vampire) [reply action-types.yml:14]: Added to `action.yml` and the types file

**Review by timheuer — APPROVED** [review]

File: action-types.yml (hunk 1 above)
Comment (LeoColman) [reply action-types.yml:14]: Great example of action typings making the documentation clearer 😄

**Pull request merged**